> Thanks for the submission! Can you add a `helm lint` call in CI as well? Also, I see a few things that were changed in the `values.yaml` which I think might cause some issues with existing template behavior.

We have already one: https://github.com/apache/superset/blob/2491b89f2911482d9951064e541d616e396f75eb/.github/workflows/superset-helm-lint.yml#L41-L45

I confirmed that `ct` lint `values.yaml` agains `values.schema.json` too:

```
$ CT_CHART_DIRS="helm" ct lint  --print-config
Linting charts...
------------------------------------------------------------------------------------------------------------------------
 Configuration
------------------------------------------------------------------------------------------------------------------------
Remote: origin
TargetBranch: master
Since: HEAD
BuildId: 
LintConf: lintconf.yaml
ChartYamlSchema: /home/adas/.ct/chart_schema.yaml
ValidateMaintainers: true
ValidateChartSchema: true
ValidateYaml: true
AdditionalCommands: []
CheckVersionIncrement: true
ProcessAllCharts: false
Charts: []
ChartRepos: []
ChartDirs: [helm]
ExcludedCharts: []
HelmExtraArgs: 
HelmRepoExtraArgs: []
Debug: false
Upgrade: false
SkipMissingValues: false
Namespace: 
ReleaseLabel: 
ExcludeDeprecated: false
------------------------------------------------------------------------------------------------------------------------

------------------------------------------------------------------------------------------------------------------------
 Charts to be processed:
------------------------------------------------------------------------------------------------------------------------
 superset => (version: "0.5.5", path: "helm/superset")
------------------------------------------------------------------------------------------------------------------------

Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "bitnami" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 2 charts
Downloading postgresql from repo https://charts.bitnami.com/bitnami
Downloading redis from repo https://charts.bitnami.com/bitnami
Deleting outdated charts
Linting chart 'superset => (version: "0.5.5", path: "helm/superset")'
Checking chart 'superset => (version: "0.5.5", path: "helm/superset")' for a version bump...
Old chart version: 0.5.4
New chart version: 0.5.5
Chart version ok.
Validating /home/adas/Devel/superset/helm/superset/Chart.yaml...
Validation success! 👍
Validating maintainers...
==> Linting helm/superset
[INFO] Chart.yaml: icon is recommended
[ERROR] values.yaml: - supersetWorker.forceReload: Invalid type. Expected: boolean, given: array

[ERROR] templates/: values don't meet the specifications of the schema(s) in the following chart(s):
superset:
- supersetWorker.forceReload: Invalid type. Expected: boolean, given: array


Error: 1 chart(s) linted, 1 chart(s) failed
------------------------------------------------------------------------------------------------------------------------
 ✖︎ superset => (version: "0.5.5", path: "helm/superset") > Error waiting for process: exit status 1
------------------------------------------------------------------------------------------------------------------------
Error: Error linting charts: Error processing charts
Error linting charts: Error processing charts
```

I confirmed also that it successfully running on CI, so I wrote in the original pull request description:

> For the purpose, I keep in that PR two commits to ensure that CI detect mismatch `values.yaml` to `values.schema.json` thanks to `.github/workflows/superset-helm-lint.yml`. If it turns out otherwise - I will update that PR.

--- a/helm/superset/Chart.yaml
+++ b/helm/superset/Chart.yaml
@@ -22,7 +22,7 @@ maintainers:
   - name: craig-rueda
     email: craig@craigrueda.com
     url: https://github.com/craig-rueda
-version: 0.5.4
+version: 0.5.5
 dependencies:
 - name: postgresql
   version: 10.2.0

--- a/helm/superset/values.schema.json
+++ b/helm/superset/values.schema.json
@@ -1,0 +1,596 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+        "replicaCount": {
+            "type": "integer"
+        },
+        "runAsUser": {
+            "type": "integer"
+        },
+        "serviceAccount": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "create": {
+                    "type": "boolean"
+                }
+            },
+            "required": [
+                "create"
+            ]
+        },
+        "bootstrapScript": {
+            "type": "string"
+        },
+        "configFromSecret": {
+            "type": "string"
+        },
+        "envFromSecret": {
+            "type": "string"
+        },
+        "envFromSecrets": {
+            "type": "array"
+        },
+        "extraEnv": {
+            "type": "object"
+        },
+        "extraEnvRaw": {
+            "type": "array"
+        },
+        "extraSecretEnv": {
+            "type": "object"
+        },
+        "extraConfigs": {
+            "type": "object"
+        },
+        "extraSecrets": {
+            "type": "object"
+        },
+        "extraVolumes": {
+            "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.api.apps.v1.PodSpec/properties/volumes"
+        },
+        "extraVolumeMounts": {
+            "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.api.apps.v1.PodSpec/properties/volumeMounts"
+        },
+        "configOverrides": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "extend_timeout": {
+                    "type": "string"
+                },
+                "enable_oauth": {
+                    "type": "string"
+                }
+            }
+        },
+        "configOverridesFiles": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "extend_timeout": {
+                    "type": "string"
+                },
+                "enable_oauth": {
+                    "type": "string"
+                }
+            }
+        },
+        "configMountPath": {
+            "type": "string"
+        },
+        "extraConfigMountPath": {
+            "type": "string"
+        },
+        "image": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "repository": {
+                    "type": "string"
+                },
+                "tag": {
+                    "type": "string"
+                },
+                "pullPolicy": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.api.core.v1.Container/properties/imagePullPolicy"
+                }
+            },
+            "required": [
+                "repository",
+                "tag",
+                "pullPolicy"
+            ]
+        },
+        "imagePullSecrets": {
+            "type": "array"
+        },
+        "service": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "type": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.api.core.v1.ServiceSpec/properties/type"
+                },
+                "port": {
+                    "type": "integer"
+                },
+                "annotations": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta/properties/annotations"
+                },
+                "loadBalancerIP": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.api.core.v1.ServiceSpec/properties/loadBalancerIP"
+                }
+            },
+            "required": [
+                "type",
+                "port"
+            ]
+        },
+        "ingress": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "enabled": {
+                    "type": "boolean"
+                },
+                "annotations": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta/properties/annotations"
+                },
+                "path": {
+                    "type": "string"
+                },
+                "pathType": {
+                    "type": "string"
+                },
+                "hosts": {
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                },
+                "tls": {
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "additionalProperties": false,
+                        "properties": {
+                            "secretName": {
+                                "type": "string"
+                            },
+                            "hosts": {
+                                "type": "array",
+                                "items": {
+                                    "type": "string"
+                                }
+                            }
+                        }
+                    }
+                }
+            },
+            "required": [
+                "enabled",
+                "annotations",
+                "path",
+                "pathType",
+                "hosts",
+                "tls"
+            ]
+        },
+        "resources": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "limits": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "properties": {
+                        "cpu": {
+                            "type": "string"
+                        },
+                        "memory": {
+                            "type": "string"
+                        }
+                    }
+                }
+            }
+        },
+        "hostAliases": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "additionalProperties": false,
+                "properties": {
+                    "ip": {
+                        "type": "string"
+                    },
+                    "hostnames": {
+                        "type": "array",
+                        "items": {
+                            "type": "string"
+                        }
+                    }
+                }
+            }
+        },
+        "supersetNode": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "command": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.api.core.v1.Container/properties/command"
+                },
+                "connections": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "properties": {
+                        "redis_host": {
+                            "type": "string"
+                        },
+                        "redis_port": {
+                            "type": "string"
+                        },
+                        "db_host": {
+                            "type": "string"
+                        },
+                        "db_port": {
+                            "type": "string"
+                        },
+                        "db_user": {
+                            "type": "string"
+                        },
+                        "db_pass": {
+                            "type": "string"
+                        },
+                        "db_name": {
+                            "type": "string"
+                        }
+                    },
+                    "required": [
+                        "redis_host",
+                        "redis_port",
+                        "db_host",
+                        "db_port",
+                        "db_user",
+                        "db_pass",
+                        "db_name"
+                    ]
+                },
+                "env": {
+                    "type": "object"
+                },
+                "forceReload": {
+                    "type": "boolean"
+                },
+                "initContainers": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.api.core.v1.PodSpec/properties/initContainers"
+                },
+                "deploymentAnnotations": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta/properties/annotations"
+                },
+                "podAnnotations": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta/properties/annotations"
+                }
+            },
+            "required": [
+                "command",
+                "connections",
+                "env",
+                "forceReload",
+                "deploymentAnnotations",
+                "podAnnotations"
+            ]
+        },
+        "supersetWorker": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "command": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.api.core.v1.Container/properties/command"
+                },
+                "forceReload": {
+                    "type": "boolean"
+                },
+                "initContainers": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.api.core.v1.PodSpec/properties/initContainers"
+                },
+                "deploymentAnnotations": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta/properties/annotations"
+                },
+                "podAnnotations": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta/properties/annotations"
+                }
+            },
+            "required": [
+                "command",
+                "forceReload",
+                "deploymentAnnotations",
+                "podAnnotations"
+            ]
+        },
+        "supersetCeleryBeat": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "enabled": {
+                    "type": "boolean"
+                },
+                "command": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.api.core.v1.Container/properties/command"
+                },
+                "forceReload": {
+                    "type": "boolean"
+                },
+                "initContainers": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.api.core.v1.PodSpec/properties/initContainers"
+                },
+                "deploymentAnnotations": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta/properties/annotations"
+                },
+                "podAnnotations": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta/properties/annotations"
+                }
+            },
+            "required": [
+                "enabled",
+                "command",
+                "forceReload",
+                "deploymentAnnotations",
+                "podAnnotations"
+            ]
+        },
+        "init": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "resources": {
+                    "type": "object"
+                },
+                "command": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.api.core.v1.Container/properties/command"
+                },
+                "enabled": {
+                    "type": "boolean"
+                },
+                "loadExamples": {
+                    "type": "boolean"
+                },
+                "createAdmin": {
+                    "type": "boolean"
+                },
+                "adminUser": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "properties": {
+                        "username": {
+                            "type": "string"
+                        },
+                        "firstname": {
+                            "type": "string"
+                        },
+                        "lastname": {
+                            "type": "string"
+                        },
+                        "email": {
+                            "type": "string"
+                        },
+                        "password": {
+                            "type": "string"
+                        }
+                    },
+                    "required": [
+                        "username",
+                        "firstname",
+                        "lastname",
+                        "email",
+                        "password"
+                    ]
+                },
+                "initContainers": {
+                    "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.api.core.v1.PodSpec/properties/initContainers"
+                },
+                "initscript": {
+                    "type": "string"
+                },
+                "podAnnotations": {
+                    "type": "object"
+                }
+            },
+            "required": [
+                "resources",
+                "command",
+                "enabled",
+                "loadExamples",
+                "createAdmin",
+                "adminUser",
+                "initscript",
+                "podAnnotations"
+            ]
+        },
+        "postgresql": {
+            "type": "object",
+            "additionalProperties": true,
+            "properties": {
+                "enabled": {
+                    "type": "boolean"
+                },
+                "existingSecret": {
+                    "type": "string"
+                },
+                "existingSecretKey": {
+                    "type": "string"
+                },
+                "service": {
+                    "type": "object",
+                    "additionalProperties": true,
+                    "properties": {
+                        "port": {
+                            "type": "integer"
+                        }
+                    },
+                    "required": [
+                        "port"
+                    ]
+                },
+                "postgresqlUsername": {
+                    "type": "string"
+                },
+                "postgresqlPassword": {
+                    "type": "string"
+                },
+                "postgresqlDatabase": {
+                    "type": "string"
+                },
+                "persistence": {
+                    "type": "object",
+                    "additionalProperties": true,
+                    "properties": {
+                        "enabled": {
+                            "type": "boolean"
+                        },
+                        "accessModes": {
+                            "type": "array",
+                            "items": [
+                                {
+                                    "type": "string"
+                                }
+                            ]
+                        }
+                    },
+                    "required": [
+                        "enabled",
+                        "accessModes"
+                    ]
+                }
+            },
+            "required": [
+                "enabled",
+                "existingSecret",
+                "existingSecretKey",
+                "service",
+                "postgresqlUsername",
+                "postgresqlPassword",
+                "postgresqlDatabase",
+                "persistence"
+            ]
+        },
+        "redis": {
+            "type": "object",
+            "additionalProperties": true,
+            "properties": {
+                "enabled": {
+                    "type": "boolean"
+                },
+                "usePassword": {
+                    "type": "boolean"
+                },
+                "existingSecret": {
+                    "type": "string"
+                },
+                "existingSecretKey": {
+                    "type": "string"
+                },
+                "password": {
+                    "type": "string"
+                },
+                "master": {
+                    "type": "object",
+                    "additionalProperties": true,
+                    "properties": {
+                        "persistence": {
+                            "type": "object",
+                            "additionalProperties": true,
+                            "properties": {
+                                "enabled": {
+                                    "type": "boolean"
+                                },
+                                "accessModes": {
+                                    "type": "array",
+                                    "items": [
+                                        {
+                                            "type": "string"
+                                        }
+                                    ]
+                                }
+                            },
+                            "required": [
+                                "enabled",
+                                "accessModes"
+                            ]
+                        }
+                    },
+                    "required": [
+                        "persistence"
+                    ]
+                },
+                "cluster": {
+                    "type": "object",
+                    "additionalProperties": true,
+                    "properties": {
+                        "enabled": {
+                            "type": "boolean"
+                        }
+                    },
+                    "required": [
+                        "enabled"
+                    ]
+                }
+            },
+            "required": [
+                "enabled",
+                "usePassword",
+                "existingSecret",
+                "existingSecretKey",
+                "password",
+                "master",
+                "cluster"
+            ]
+        },
+        "nodeSelector": {
+            "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.api.apps.v1.PodSpec/properties/nodeSelector"
+        },
+        "tolerations": {
+            "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json##/definitions/io.k8s.api.apps.v1.PodSpec/properties/tolerations"
+        },
+        "affinity": {
+            "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.23.0/_definitions.json#/definitions/io.k8s.api.core.v1.Affinity"
+        }
+    },
+    "required": [
+        "replicaCount",
+        "runAsUser",
+        "serviceAccount",
+        "bootstrapScript",
+        "configFromSecret",
+        "envFromSecret",
+        "envFromSecrets",
+        "extraEnv",
+        "extraEnvRaw",
+        "extraSecretEnv",
+        "extraConfigs",
+        "extraSecrets",
+        "extraVolumes",
+        "extraVolumeMounts",
+        "configOverrides",
+        "configOverridesFiles",
+        "configMountPath",
+        "extraConfigMountPath",
+        "image",
+        "imagePullSecrets",
+        "service",
+        "ingress",
+        "resources",
+        "hostAliases",
+        "supersetNode",
+        "supersetWorker",
+        "supersetCeleryBeat",
+        "init",
+        "postgresql",
+        "redis",
+        "nodeSelector",
+        "tolerations",
+        "affinity"
+    ]
+}

--- a/helm/superset/values.yaml
+++ b/helm/superset/values.yaml
@@ -149,7 +149,7 @@ configOverrides: {}
   #   # The default user self registration role
   #   AUTH_USER_REGISTRATION_ROLE = "Admin"
 # Same as above but the values are files
-configOverridesFiles:
+configOverridesFiles: {}
   # extend_timeout: extend_timeout.py
   # enable_oauth: enable_oauth.py
 

--- a/helm/superset/values.yaml
+++ b/helm/superset/values.yaml
@@ -357,7 +357,7 @@ postgresql:
   enabled: true
   ##
   ## The name of an existing secret that contains the postgres password.
-  existingSecret:
+  existingSecret: secret-name
   ## Name of the key containing the secret.
   existingSecretKey: postgresql-password
   ##
@@ -402,7 +402,7 @@ redis:
   usePassword: false
   ##
   ## The name of an existing secret that contains the redis password.
-  existingSecret:
+  existingSecret: secret-name
   ## Name of the key containing the secret.
   existingSecretKey: redis-password
   ##

